base_user_role_company/models/ir_http.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.http import request
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = "ir.http"
+
+    def session_info(self):
+        """
+        Based on the selected companies (cids),
+        calculate the roles to enable.
+        A role should be enabled only when it applies to all selected companies.
+        """
+        result = super().session_info()
+        if self.env.user.role_line_ids:
+            cids_str = request.httprequest.cookies.get("cids", str(self.env.company.id))
+            cids = [int(cid) for cid in cids_str.split(",")]
```

Comment:
It seems cids are split by dashes in the cookie as of https://github.com/odoo/odoo/commit/e8d80ceb348#diff-747c1e41e030ec3b083ce9f5405cd3a0f950656ad6a59dbfc39e744955b81472.
If I enable more than one company on a user with roles, I get
```
  File "/mnt/data/odoo-addons-dir/base_user_role_company/models/ir_http.py", line 20, in session_info
    cids = [int(cid) for cid in cids_str.split(",")]
  File "/mnt/data/odoo-addons-dir/base_user_role_company/models/ir_http.py", line 20, in <listcomp>
    cids = [int(cid) for cid in cids_str.split(",")]
ValueError: invalid literal for int() with base 10: '2-1'
```

Reply:
Yep, agree with you we should split companies' ids by dashes.

Reply:
Thanks, that seems fixed now.